### PR TITLE
Add condition expression assertion for ExpectedJoinTable.

### LINKED ISSUE #7880: Add UT for table and expr.
## Background
Now, table and expr in statement  are improved, so origin UT for it is not suitable. We need to add UT for table and expr.
parser UT entry:`shardingsphere/shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/engine/SQLParserParameterizedTest.java->assertSupportedSQL()`

## Solutions
1. Think about design of xxxAssert.java.
2. change expected result.(path:`shardingsphere/shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/resources/case`)
3. The test class runs successfully(`shardingsphere/shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/engine/SQLParserParameterizedTest.java).

### eg
* #7872 

## Note
We can discuss the reasonableness of the design and then implement it.

### subtasks

- [x]  add UT for table
- [x]  add UT for expr

### PR DESCRIPTION
**Pull request by lwtdev**
Fixes #7880.

Changes in this pull request:

  * Add `condition` and `columns` assertions for `ExpectedJoinTable`.
  * Adjust the old test case, replace joinSpecification with condition or columns in `join-table`.
  * Delete unused expected jaxb classes(such as `ExpectedJoinSpecification.java`, `ExpectedAndPredicate.java`).
  * Move `ExpectedWhereClause.java` from `impl.predicate` to `impl.where`.

### CODE REVIEW

**Review by terrymanu — CHANGES_REQUESTED** [review]
It is unnecessary to change the package name, can you revert that?

**Comment by lwtdev** [comment]
> It is unnecessary to change the package name, can you revert that?

@terrymanu 

Rename `expr` to `expression` is in order to be consistent with the name under `asserts.segment` (`org.apache.shardingsphere.test.sql.parser.parameterized.asserts.segment.expression`).

**Comment by terrymanu** [comment]
> > It is unnecessary to change the package name, can you revert that?
> 
> @terrymanu
> 
> Rename `impl.expr` to `impl.expression` is in order to be consistent with the name under `asserts.segment` (`org.apache.shardingsphere.test.sql.parser.parameterized.asserts.segment.expression`).
> 
> Maybe I should rename `asserts.segment.expression` to `asserts.segment.expr`?
> Becasue `ExpressionSegment` is under package `org.apache.shardingsphere.sql.parser.sql.common.segment.dml.expr`.

If rename is necessary, it is better to split them to another pr.
It is easy to review if one pr do one thing only.

**Comment by lwtdev** [comment]
> > > It is unnecessary to change the package name, can you revert that?
> > 
> > 
> > @terrymanu
> > Rename `impl.expr` to `impl.expression` is in order to be consistent with the name under `asserts.segment` (`org.apache.shardingsphere.test.sql.parser.parameterized.asserts.segment.expression`).
> > Maybe I should rename `asserts.segment.expression` to `asserts.segment.expr`?
> > Becasue `ExpressionSegment` is under package `org.apache.shardingsphere.sql.parser.sql.common.segment.dml.expr`.
> 
> If rename is necessary, it is better to split them to another pr.
> It is easy to review if one pr do one thing only.

I will revert it~

File: shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/segment/impl/table/ExpectedJoinTable.java [hunk 1]
```diff
@@ -20,23 +20,29 @@
...
     private ExpectedTable right;
-    
-    @XmlElement
-    private ExpectedJoinSpecification joinSpecification;
+
+    @XmlElement(name = "condition")
+    private ExpectedExpression condition;
+
+    @XmlElement(name = "columns")
+    private final List<ExpectedColumn> columns = new LinkedList<>();
```
Comment (jingshanglu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/segment/impl/table/ExpectedJoinTable.java:47]: @lwtdev Maybe it is better to rename `columns` to `using-columns`.

Comment (lwtdev) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/segment/impl/table/ExpectedJoinTable.java:47]: @jingshanglu OK, I will modify it. What about `condition`, do we should rename it to `on-condition`?

Comment (jingshanglu) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/segment/impl/table/ExpectedJoinTable.java:47]: @lwtdev I think both will do.

**Review by jingshanglu — APPROVED** [review]

**Comment by jingshanglu** [comment]
> It is unnecessary to change the package name, can you revert that?

@terrymanu It is a `change request` review, need you merge the pr.

**Review by terrymanu — APPROVED** [review]

**Pull request merged**